From a4037730e68cfc37288badc64f4ddaf6c1fa51e8 Mon Sep 17 00:00:00 2001
From: Christian Hergert <chergert@redhat.com>
Date: Wed, 27 Apr 2016 16:51:25 -0700
Subject: [PATCH] iconhelper: check if surface was invalidated

Fetching the style may cause the surface to be invalidated.

We don't want to fetch the style unless we already have a surface,
so just do a second surface check after fetching the style.
---
 gtk/gtkiconhelper.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/gtk/gtkiconhelper.c b/gtk/gtkiconhelper.c
index b8737e7..a80418d 100644
--- gtk/gtkiconhelper.c
+++ gtk/gtkiconhelper.c
@@ -896,10 +896,16 @@ _gtk_icon_helper_draw (GtkIconHelper *self,
 
   if (self->priv->rendered_surface != NULL)
     {
-      gtk_css_style_render_icon_surface (gtk_css_node_get_style (gtk_css_gadget_get_node (GTK_CSS_GADGET (self))),
-                                         cr,
-                                         self->priv->rendered_surface,
-                                         x, y);
+      GtkCssStyle *style;
+
+      style = gtk_css_node_get_style (gtk_css_gadget_get_node (GTK_CSS_GADGET (self)));
+
+      /* Fetching the style might cause rendered_surface to be invalidated */
+      if (self->priv->rendered_surface != NULL)
+        gtk_css_style_render_icon_surface (style,
+                                           cr,
+                                           self->priv->rendered_surface,
+                                           x, y);
     }
 }
 
-- 
2.3.2 (Apple Git-55)

